eunkyung/app.js

```diff
@@ -34,25 +34,141 @@ app.get('/ping', (req, res) => {
                 ?,
                 ?
                 );`,
-        [name, password, email]
+        [name, password, email, profile_image]
     )
     res.status(201).json({ message: 'userCreated' })
 })
 
 
 
+app.post('/posting', async (req, res) => {
+    const { title, content, post_image_url, user_id } = req.body
+    await mysqlDataSource.query(
+        `INSERT INTO posts (
+            title,
+            content,
+            post_image_url,
+            user_id
+        )
+        VALUES (
+            ?,
+            ?,
+            ?,
+            ?
+        );`,
+        [title, content, post_image_url, user_id]
+    )
+    res.status(201).json({ message: 'postCreated' })
+})
+
+app.get('/posts', async (req, res) => {
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        posts.user_id as postingId,
+        posts.post_image_url as postingImageUrl,
+        posts.content as postingContent 
+        FROM users INNER JOIN posts ON users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+
+app.get('/posts/:userId', async (req, res) => {
+    const { userId } = req.params;
+
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        pp.postings
+        FROM users
+        LEFT JOIN (
+            SELECT 
+            user_id,
+            JSON_ARRAYAGG(
+                JSON_OBJECT(
+                    'postingId', posts.id,
+                    'postingImageUrl', posts.post_image_url,
+                    'postingContent', posts.content
+                ) 
+            ) 
+        as postings
+        FROM posts 
+        GROUP BY user_id  
+        ) pp ON users.id = pp.user_id
+            WHERE users.id = ${userId};`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.patch('/edit/:userId/:postId', async (req, res) => {
+    const { userId, postId } = req.params;
+    const { postingContent } = req.body
+
+    await mysqlDataSource.query(
+        `UPDATE posts 
+        SET 
+        posts.content = ? 
+        WHERE posts.user_id = ${userId} AND posts.id = ${postId}
+        `,
+        [postingContent])
+
+
+    await mysqlDataSource.query(
+        `SELECT users.id as userId, users.name as userName, posts.id as postingId, posts.title as postingTitle, posts.content as postingContent FROM users INNER JOIN posts ON users.id = ${userId} AND posts.id = ${postId} WHERE users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.delete('/delete/:postId', async (req, res) => {
+    const { postId } = req.params;
+
+    await mysqlDataSource.query(
+        `DELETE FROM posts WHERE posts.id = ${postId}`
+    )
+    res.status(201).json({ message: 'postingDeleted' })
+})
```

Comment:
해당 부분도 마찬가지로 `delete` 라는 Method에 삭제하겠다는 의미가 내포되어 있으므로 엔드포인트 url에 delete를 명시하지 않아도 됩니다!
```suggestion
app.delete('/:postId', async (req, res) => {
    const { postId } = req.params;

    await mysqlDataSource.query(
        `DELETE FROM posts WHERE posts.id = ${postId}`
    )
    res.status(201).json({ message: 'postingDeleted' })
})
```